Objects/genericaliasobject.c

```diff
@@ -244,6 +247,48 @@ make_parameters(PyObject *args)
     return parameters;
 }
 
+/* If obj is a generic alias, substitute type variables params
+   with substitutions argitems. */
+static PyObject *
+subs_tvars(PyObject *obj, PyObject *params, PyObject **argitems)
+{
+    _Py_IDENTIFIER(__parameters__);
+    PyObject *subparams;
+    if (_PyObject_LookupAttrId(obj, &PyId___parameters__, &subparams) < 0) {
+        return NULL;
+    }
+    if (subparams && PyTuple_Check(subparams)) {
+        Py_ssize_t nparams = PyTuple_GET_SIZE(params);
```

Comment:
This could be moved into the `if (nsubargs != 0) {` block below.

(Also the logic would be slightly simpler if the check for subparams being non-empty was part of the `if (subparams && PyTuple_Check(subparams)) {` check above.)